src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -791,15 +791,41 @@ def _validate_lock_params_match_lock(
                 name, _resource_name))
 
 
-def get_lock(name, resource_group_name=None):
+def get_lock(name, resource_group_name=None, resource_provider_namespace=None,
+             parent_resource_path=None, resource_type=None, resource_name=None):
     """
-    :param name: Name of the lock.
+    :param name: The name of the lock.
     :type name: str
+    :param resource_provider_namespace: Name of a resource provider.
+    :type resource_provider_namespace: str
+    :param parent_resource_path: Path to a parent resource
+    :type parent_resource_path: str
+    :param resource_type: The type for the resource with the lock.
+    :type resource_type: str
+    :param resource_name: Name of a resource that has a lock.
+    :type resource_name: str
     """
     lock_client = _resource_lock_client_factory()
+
+    lock_resource = _extract_lock_params(resource_group_name, resource_provider_namespace,
```

Comment:
I believe `_extract_lock_params` may be the source of the bug that doesn't allow `--resource-type` to accept <NAMESPACE/TYPE>